pyomo/contrib/pynumero/tests/test_cyipopt_examples.py

```diff
@@ -149,3 +153,28 @@ def test_parameter_estimation(self):
 
         m = ex.perform_estimation_pyomo_only(data_fname, solver_trace=False)
         self.assertAlmostEqual(pyo.value(m.UA), 204.43761, places=3)
+
+    def test_cyipopt_callbacks(self):
+        ex = import_file(os.path.join(example_dir, 'callback', 'cyipopt_callback.py'))
+
+        output = StringIO()
+        with LoggingIntercept(output, 'pyomo', logging.INFO):
+            ex.main()
+
+        self.assertIn("Residuals for iteration 2",
+                          output.getvalue().strip())
+
+    @unittest.skipIf(not pandas_available, "pandas needed to run this example")
+    def test_cyipopt_functor(self):
+        ex = import_file(os.path.join(example_dir, 'callback', 'cyipopt_functor_callback.py'))
+        df = ex.main()
+        self.assertTrue(df.shape == (7, 5))
```

Comment:
I generally prefer
```python
self.assertEqual(df.shape, (7, 5))
```
because the error message is more informative if the test ever fails.

Reply:
done.